api/v1alpha1/cdapmaster_types.go

```diff
@@ -124,6 +129,9 @@ type CDAPServiceSpec struct {
 	SecretVolumes map[string]string `json:"secretVolumes,omitempty"`
 	// SecurityContext overrides the security context for the service pods.
 	SecurityContext *SecurityContext `json:"securityContext,omitempty"`
+	// DisableSystemMetricsSidecar allows service to disable system metrics sidecar container
+	// even if SystemMetricsExporter spec is present.
+	DisableSystemMetricsSidecar *bool `json:"disableSystemMetricsSidecar,omitempty"`
```

Comment:
Better to make it "EnableMetricCollection" that can default to true.

Reply:
I've assumed missing/null as true as a way to have "default true" behaviour.  https://book.kubebuilder.io/reference/markers/crd-validation.html#crd-validation describes an annotation for setting default, but we'll need to update k8s api being used from v1beta1 to v1.
If the SystemMetricsExporter spec isn't present in CR, this flag will not have any effect and no sidecars will be launched. This has the benefit of not starting sidecars in existing instances with old CDAP images and failing.

Reply:
While fetching the instance of CDAPServiceSpec in getCDAPServiceSpec() function, we can set this field to true (if absent) when SystemMetricsExporter is present. That will allow make the check more straight forward while adding side car.

Reply:
I've moved the logic to default enableSystemMetrics and check whether sidecar is enabled into the function that was adding the sidecar and simplified the condition here. Renamed to function to `addSystemMetricsServiceIfEnabled`. 
I think it's not expected behaviour that getCDAPServiceSpec will also set a default value. What do you think about the changes?